app.go

```diff
@@ -264,3 +264,56 @@ func (c *Client) RevokeTokens(ctx context.Context, before *time.Time) (*Response
 	err := c.makeRequest(ctx, http.MethodPatch, "app", nil, setting, &resp)
 	return &resp, err
 }
+
+type PushProvider struct {
+	Type           string     `json:"type"`
+	Name           string     `json:"name"`
+	Description    string     `json:"description,omitempty"`
+	DisabledAt     *time.Time `json:"disabled_at,omitempty"`
+	DisabledReason string     `json:"disabled_reason,omitempty"`
+
+	APNAuthKey              string `json:"apn_auth_key,omitempty"`
+	APNKeyID                string `json:"apn_key_id,omitempty"`
+	APNP12Cert              string `json:"apn_p12_cert,omitempty"`
+	APNTeamID               string `json:"apn_team_id,omitempty"`
+	APNTopic                string `json:"apn_topic,omitempty"`
+	APNAuthType             string `json:"apn_auth_type,omitempty"`
+	APNDevelopment          bool   `json:"apn_development,omitempty"`
+	APNHost                 string `json:"apn_host,omitempty"`
+	APNNotificationTemplate string `json:"apn_notification_template,omitempty"`
+
+	FirebaseServerKey            string `json:"firebase_server_key,omitempty"`
+	FirebaseCredentials          string `json:"firebase_credentials,omitempty"`
+	FirebaseDataTemplate         string `json:"firebase_data_template,omitempty"`
+	FirebaseNotificationTemplate string `json:"firebase_notification_template,omitempty"`
```

Comment:
we will use this one for clarity with v1, we will name differently so for this PR, drop it too